testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/rx/RxCompletionStageTest.java

```diff
@@ -0,0 +1,377 @@
+      CompletionStageRxInvoker invoker = client.target(generateURL("/put/thing")).request().rx(CompletionStageRxInvoker.class);
+      CompletionStage<Thing> completionStage = invoker.put(aEntity, Thing.class);
+      Assert.assertEquals(new Thing("a"), completionStage.toCompletableFuture().get());
+   }
+
+   @Test
+   public void testPutThingList() throws Exception {
+      CompletionStageRxInvoker invoker = client.target(generateURL("/put/thing/list")).request().rx(CompletionStageRxInvoker.class);
+      CompletionStage<List<Thing>> completionStage = invoker.put(aEntity, LIST_OF_THING);
+      Assert.assertEquals(aThingList, completionStage.toCompletableFuture().get());
+   }
+
+   @Test
+   public void testPost() throws Exception {
+      CompletionStageRxInvoker invoker = client.target(generateURL("/post/string")).request().rx(CompletionStageRxInvoker.class);
+      CompletionStage<Response> completionStage = invoker.post(aEntity);
+      Assert.assertEquals("a", completionStage.toCompletableFuture().get().readEntity(String.class));
+   }
+
+   @Test
+   public void testPostThing() throws Exception {
+      CompletionStageRxInvoker invoker = client.target(generateURL("/post/thing")).request().rx(CompletionStageRxInvoker.class);
+      CompletionStage<Thing> completionStage = invoker.post(aEntity, Thing.class);
+      Assert.assertEquals(new Thing("a"), completionStage.toCompletableFuture().get());
+   }
+
+   @Test
+   public void testPostThingList() throws Exception {
+      CompletionStageRxInvoker invoker = client.target(generateURL("/post/thing/list")).request().rx(CompletionStageRxInvoker.class);
+      CompletionStage<List<Thing>> completionStage = invoker.post(aEntity, LIST_OF_THING);
+      Assert.assertEquals(aThingList, completionStage.toCompletableFuture().get());
+   }
+
+   @Test
+   public void testDelete() throws Exception {
+      CompletionStageRxInvoker invoker = client.target(generateURL("/delete/string")).request().rx(CompletionStageRxInvoker.class);
+      CompletionStage<Response> completionStage = invoker.delete();
+      Assert.assertEquals("x", completionStage.toCompletableFuture().get().readEntity(String.class));
+   }
+
+   @Test
+   public void testDeleteThing() throws Exception {
+      CompletionStageRxInvoker invoker = client.target(generateURL("/delete/thing")).request().rx(CompletionStageRxInvoker.class);
+      CompletionStage<Thing> completionStage = invoker.delete(Thing.class);
+      Assert.assertEquals(new Thing("x"), completionStage.toCompletableFuture().get());
+   }
+
+   @Test
+   public void testDeleteThingList() throws Exception {
+      CompletionStageRxInvoker invoker = client.target(generateURL("/delete/thing/list")).request().rx(CompletionStageRxInvoker.class);
+      CompletionStage<List<Thing>> completionStage = invoker.delete(LIST_OF_THING);
+      Assert.assertEquals(xThingList, completionStage.toCompletableFuture().get());
+   }
+
+   @Test
+   @Ignore // @TODO Fix
```

Comment:
?

Reply:
@asoldano, there's a problem, maybe in undertow, such that after making a HEAD call, the connection hangs. I set it aside in RESTEASY-1885 for later attention.

Reply:
OK, no problem, then just add a reference to RESTEASY-1885 in the TODO comment please :-)

Reply:
@asoldano done. Haven't committed it yet.